packages/react-components/src/components/FileDownloadCards.tsx

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { Icon, Spinner, SpinnerSize } from '@fluentui/react';
+import React, { useCallback, useState } from 'react';
+import { _FileCard } from './FileCard';
+import { _FileCardGroup } from './FileCardGroup';
+import { extension } from './utils';
+
+/**
+ * Meta Data containing information about the uploaded file.
+ * @beta
+ */
+export interface FileMetadata {
+  /**
+   * File name to be displayed.
+   */
+  name: string;
+  /**
+   * Extension is used for rendering the file icon.
+   * An unknown extension will be rendered as a generic icon.
+   * Example: `jpeg`
+   */
+  extension: string;
+  /**
+   * Download URL for the file.
+   */
+  url: string;
+}
+
+/**
+ * @beta
+ * A file download error returned via a {@link FileDownloadHandler}.
+ * This error message is used to render an error message in the UI.
+ */
+export interface FileDownloadError {
+  /** The error message to display in the UI */
+  errorMessage: string;
+}
+
+/**
+ * @beta
+ *
+ * A callback function for handling file downloads.
+ * The function needs to return a promise that resolves to a file download URL.
+ * If the promise is rejected, the {@link Error.message} will be used to display an error message to the user.
+ *
+ * Sample Usage:
```

Comment:
nit: use [`@example`](https://tsdoc.org/pages/tags/example/) so inteliisense/docs things picks this up nicely